src/safeposix/syscalls/net_calls.rs

```diff
@@ -1567,27 +1565,34 @@ impl Cage {
                         structpoll.events |= POLLERR;
                     }
                     poll_fds_vec.push(structpoll);
-                }
-
+                  num_events += 1;
+                } 
                 let poll_fds_slice = &mut poll_fds_vec[..];
                 Self::poll_syscall(&self, poll_fds_slice, timeout);
-                let mut count_changed: i32 = 0;
-
-                for (count, result) in poll_fds_slice[..maxevents as usize].iter().enumerate() {
+                let mut count = 0;
+                let end_idx: usize = core::cmp::min(num_events as usize, maxevents as usize);
```

Comment:
we don't use outside imports directly in safeposix, instead use interface::rust_min

Reply:
I also think num_events is already usize so you dont have to cast?